### ISSUE TITLE
Unexpected Duplicate Large Edge Files and File Size Increase

### ISSUE THREAD

**Issue by ludralph** [prisma/prisma#21628]
### Bug description

A user switched to Cloudfare Workers and noticed that they have two edge files whose sizes are very large. They are wondering why they have two edge files instead of one. They also noticed an increase in the size of the edge files and want to know if they can get rid of one.

Link to Zendesk ticket [#3099](https://prisma1015.zendesk.com/agent/tickets/3099)
[Link to discussion in Slack](https://prisma-company.slack.com/archives/C016KUHB1R6/p1698305628895569)

### How to reproduce

![image](https://github.com/prisma/prisma/assets/13855608/819dec15-1c08-4620-9823-62e0a3cbe147)

### Expected behavior

On deployment, we should have one edge file with reduced bundle size

### Prisma information

N/A

### Environment & setup

Cloudfare Worker Deployment

### Prisma Version

5.4.2

**Comment by janpio** [comment]
From looking at the screenshot, I guess one is the runtime (the one in `@prisma/client/runtime/edge.js` ) and the other one is the generated Client in `.prisma/client/edge.js`). Same name, but different files.

So question back: Is the content of those files identical?

> They also noticed an increase in the size of the edge files

Increase in what context? From one version to another version? If so, which versions?

**Comment by karthikjn01** [comment]
Cheers @ludralph and @janpio for looking into this.

For some more detail, it wasn't an increase in file size switching. It was that I want to reduce the bundle size as the current _worker.js file is way too large, there is a limit on the free tier for 1mb - currently using the paid version which increases the size to 10mb. When I looked into what was causing the file size using @next/bundle-analyzer it indicated that every page that I had on the edge had the duped edge.js files from the picture above. I'm hoping that removing one or the other could potentially bring the file size down.

> So question back: Is the content of those files identical?

I don't think so as they look like they are two different sizes, and one of them doesn't contain the `withAccelerator` module.

**Comment by karthikjn01** [comment]
Oh I also forgot to mention @janpio - the cloudflare team say that if the file size of the worker.js goes beyond 1mb it can cause longer coldstarts - which is exactly what I'm trying to eliminate by switching to cloudlare.

**Comment by janpio** [comment]
Ok, that makes sense and is of course something we want to improve.

But to double check again: 
You did not observe an _increase_ in that size between two different Prisma versions?
(Then I will remove that from the issue title as it can be misleading)

**Comment by karthikjn01** [comment]
Yep, correct. I didn't notice a change in size between the versions of prisma :)

**Comment by karthikjn01** [comment]
Any thoughts/updates on this @janpio

**Comment by janpio** [comment]
What thoughts do you expect? 

We now understand that you want these files to be smaller. We will look into if we can make these file smaller when we have time for it, or this becomes a focus of ours to optimize usage of Prisma Client in edge environments.

Something you can try already, and might move things around and as side effect also lead to an overall smaller size could be to define a custom `output` folder: https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#using-a-custom-output-path (Note that the import has to be adapted to the new path!) That should copy the "runtime" from `@prisma/client` as well and allow your bundle to skip the files from `node_modules`.

**Comment by lky141319** [comment]
Claiming bounty. 
Wallet: 0xd2268DEDca16b08aF4bb6Ad20cBa877D10cf8Ac6
GitHub: @lky141319

**Comment by puneetdixit200** [comment]
Hi maintainers, I’d like to work on this bundle-size issue. I can inspect why duplicate edge files are emitted, quantify the size impact, and propose a minimal packaging fix.
